src/config.rs

```diff
@@ -0,0 +1,219 @@
+                );
+                exit(1);
+            }
+        }
+
+        let out: Config = match toml::from_str(&config_content) {
+            Err(_) => {
+                let example_config_file = format!("{}.example", config_file);
+                make_default_config(&example_config_file);
+                eprintln!("Configuration file invalid!\nAn example can be found here: {}", example_config_file);
+                exit(1);
+            },
+            Ok(config) => config,
+        };
+        out
+    }
+}
+
+/// A function to make a configuration file.
+///
+/// # Arguments
+///
+/// * `config_file`: The location where the new config file should be placed.
+///
+/// returns: ()
+///
+/// # Examples
+///
+/// ```
+/// let config_file: String = String::from("config.toml");
+/// let future = make_default_config(&config_file);
+/// block_on(future);
+/// ```
+fn make_default_config(config_file: &String) {
+    // Try to create the file
+    let mut file = match File::create(config_file) {
+        Ok(f) => f,
+        Err(why) => panic!("Could not create the config file: {:?}.\n{}\n{}", why, PANIC_RESPONSE, why)
+    };
+
+    // Write the config file
+    match writeln!(&mut file, "# Config created automatically\n{}", include_str!("../config.toml.example")) {
+        Ok(_) => println!("Written default configuration to {}", config_file),
+        Err(why) => panic!("Could not write default configuration file to {}: {:?}\n{}\n{}", config_file, why, PANIC_RESPONSE, why)
+    };
+}
+
+/// Checks if the inputted config is compatible with the program
+///
+/// # Arguments
+///
+/// * `config_content`: The content of the config file to check for compatibility
+///
+/// returns: (bool, String)
+///
+/// # Examples
+///
+/// ```
+/// if ! check_version("config.toml")[0] {
+///     panic!("Config version incompatible!");
+/// }
+/// ```
+fn check_version(config_content: &String) -> (bool, String) {
+    // Get the version, by looping over the config contents, searching for the config line and
+    // getting the version part of it.
+    let version = {
+        let version_line = Regex::new("(?m)^version *= *\"(?P<major>\\d*)\\.(?P<minor>\\d*)\" *(|#.*)$").unwrap();
+        let captures = version_line.captures(config_content).unwrap();
+        let out = format!("{}.{}", captures.name("major").unwrap().as_str(), captures.name("minor").unwrap().as_str());
+        dbg!("{}", &out);
+        out
+    };
+    // Config file is always compatible with its associated program version
+    if format!("{}.{}", env!("CARGO_PKG_VERSION_MAJOR"), env!("CARGO_PKG_VERSION_MINOR")).eq_ignore_ascii_case(&version) {
+        return (true, version);
+    }
+    // If config file version is not the program version, check if it is still compatible:
+    for (major, minor) in COMPATIBLE_VERSIONS {
+        if format!("{}.{}", major, minor).eq_ignore_ascii_case(&version) {
+            return (true, version);
+        }
+    }
+    return (false, version);
+}
```

Comment:
Use single matching group, accept any string slice, return string slice
```suggestion
fn check_version(config_content: &str) -> (bool, &str) {
    // Get the version, by searching for the config line and getting the version part of it.
    let version = {
        let captures = VERSION_REGEX.captures(config_content).unwrap();
        let out = captures.name("version").unwrap().as_str();
        dbg!("{}", out);
        out
    };
    // Config file is always compatible with its associated program version
    if format!("{}.{}", env!("CARGO_PKG_VERSION_MAJOR"), env!("CARGO_PKG_VERSION_MINOR")).eq_ignore_ascii_case(version) {
        return (true, version);
    }
    // If config file version is not the program version, check if it is still compatible:
    for (major, minor) in COMPATIBLE_VERSIONS {
        if format!("{}.{}", major, minor).eq_ignore_ascii_case(version) {
            return (true, version);
        }
    }
    return (false, version);
}
```